tests/python/relax/test_transform_rewrite_cuda_graph.py

```diff
@@ -709,7 +709,7 @@ def main():
 def test_static_args():
```

Comment:
This test case seemed a little concerning to me because the transformation hides the potentially impure packed call in `Before`'s `main` function and also because the `builtin_with_ctx` is not considered pure (maybe we should add a pure override for cases like that)?

Reply:
Agreed, I feel like the `main` function should be marked as impure, because it entirely consists of side effects.

I could see a case for having a wrapper function marked as pure, even if it calls impure functions.  That's kind of what the `R.call_tir` builtin does, to provide a pure wrapper for an impure TIR function.  However, that wouldn't be the case here, as it the `cuda_graph.run_or_capture` function is entirely impure.